src/Uno.UI.Tests/Windows_UI_Xaml/Given_ResourceDictionary.cs

```diff
@@ -795,5 +795,20 @@ public void When_Custom_Resource_Dictionary_With_Custom_Property_in_Custom_Contr
 			Assert.IsTrue(resources.ContainsKey("TestKey"));
 			Assert.AreEqual(resources["TestKey"], "Test123");
 		}
+
+		[TestMethod]
+		public void When_Theme_Dictionary_Is_Cached()
+		{
+			var dictionary = new ResourceDictionary();
+			dictionary.TryGetValue("TestKey", out _); // This causes _activeThemeDictionary to be cached.
+
+			var lightTheme = new ResourceDictionary();
+			lightTheme.Add("TestKey", "TestValue");
+			dictionary.ThemeDictionaries.Add("Light", lightTheme); // Cached value is no longer valid.
```

Comment:
I'm not sure hard coding "Light" here is correct.

Reply:
You could use a nameof from the RequestedTheme, but that's unlikely to change.